### PR TITLE
Portfolio Project: CSS Grid Layout Challenge

### PR DESCRIPTION
**Pull request by laurosilvacom**
[Preview](https://egghead-io-nextjs-git-zj-ls-css-grid-project-eggheadio1.vercel.app/projects/build-modern-layouts-with-css-grid)

image

### CODE REVIEW

**Comment by joelhooks** [comment]
My suggestion is to build this more like a recipe. It's effectively a "blog post" in structure with clear linear instructions so this presentation doesn't feel natural. As a user, I want to be initially aligned and oriented with the page's purpose and the outcome it will bring for me. 

A hook that let's me know what's in it for me followed by the recipe with clear next steps and expected outcomes.

Initially this is a copywriting puzzle imo.

I'd also love to discuss how they get to this page 🤔

File: src/pages/projects/build-modern-layouts-with-css-grid/index.tsx [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+  const dashboardPageChallenge: any = find(course?.projects?.content, {
+    label: 'Dashboard page challenge',
+  })
+  const studyQuestions: any = find(course?.projects?.content, {
+    label: 'Self-Study Questions',
+  })
+  const bestPractices: any = find(course?.projects?.content, {
+    label: 'Best Practices',
+  })
+  const Submission: any = find(course?.projects?.content, {label: 'Submission'})
```
Comment (zacjones93) [inline src/pages/projects/build-modern-layouts-with-css-grid/index.tsx:51]: Sanity returns an array of content so accessing that data through lodash `find` so we don't have to worry about it in the view

File: src/pages/projects/build-modern-layouts-with-css-grid/index.tsx [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+const landingPage: FunctionComponent<LandingProps> = (props) => {
+  const {course} = props
+
+  const {
+    pricingPageFigmaUrl,
+    crmPageFigmaUrl,
+    pricingPageImageUrl,
+    crmPageImageUrl,
+    ogImage,
+  } = course?.projects
```
Comment (zacjones93) [inline src/pages/projects/build-modern-layouts-with-css-grid/index.tsx:22]: destructuring the urls we are using for this project

File: src/pages/projects/build-modern-layouts-with-css-grid/index.tsx [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+            <h1 className="text-2xl md:text-4xl font-bold mt-2">
+              {course.projects.title}
+            </h1>
+          </div>
+          <div className="flex flex-col justify-center items-start mx-auto max-w-screen-md mb-16 w-full px-4">
+            <main className="prose dark:prose-dark prose-lg max-w-none w-full">
+              <Markdown>{introduction.text}</Markdown>
+              <Markdown>{challenges.text}</Markdown>
+
+              <section className="relative dark:text-gray-200 mt-16 mb-16 ml-0 mr-0 md:-ml-20 md:-mr-20">
```
Comment (zacjones93) [inline src/pages/projects/build-modern-layouts-with-css-grid/index.tsx:94]: added the challenges section to display the 2 different layouts side by side with description text

File: src/pages/projects/build-modern-layouts-with-css-grid/index.tsx [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+              Portfolio Project
+            </p>
+            <h1 className="text-2xl md:text-4xl font-bold mt-2">
+              {course.projects.title}
+            </h1>
+          </div>
+          <div className="flex flex-col justify-center items-start mx-auto max-w-screen-md mb-16 w-full px-4">
+            <main className="prose dark:prose-dark prose-lg max-w-none w-full">
+              <Markdown>{introduction.text}</Markdown>
+              <Markdown>{challenges.text}</Markdown>
```
Comment (zacjones93) [inline src/pages/projects/build-modern-layouts-with-css-grid/index.tsx:92]: most content is just being rendered as markdown

File: src/pages/projects/build-modern-layouts-with-css-grid/index.tsx [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+              <Markdown>{outcomes.text}</Markdown>
+              <section className="relative dark:text-gray-200 mt-16 mb-16 ml-0 mr-0 md:-ml-28 md:-mr-28">
+                <div className="grid grid-cols-1 lg:grid-cols-2 mt-16 mb-16 gap-5">
+                  <div className="mb-8 md:mb-0 text-center relative">
+                    <div className="bg-white dark:bg-gray-800 rounded-xl text-center p-4 md:p-8 z-20 relative shadow-lg">
+                      <Markdown className="prose dark:prose-dark pt-2 sm:text-base text-sm leading-normal text-gray-800 dark:text-gray-200 mt-0 text-left">
+                        {pricingPageChallenge.text}
+                      </Markdown>
+                    </div>
+                  </div>
```
Comment (zacjones93) [inline src/pages/projects/build-modern-layouts-with-css-grid/index.tsx:175]: breaking up the content with cards for the challenge requirements

File: src/pages/projects/build-modern-layouts-with-css-grid/index.tsx [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+                className="inline-flex justify-center items-center px-6 py-4 font-semibold rounded-md bg-blue-600 text-white transition-all hover:bg-blue-700 ease-in-out duration-200"
+                title="Share on twitter"
+                href={course.projects.tweetCTA}
+                rel="noopener"
+                style={{
+                  color: 'white',
+                  textDecoration: 'none',
+                }}
+              >
+                Tweet @eggheadio
```
Comment (zacjones93) [inline src/pages/projects/build-modern-layouts-with-css-grid/index.tsx:200]: keeping the tweet egghead cta

File: src/pages/projects/build-modern-layouts-with-css-grid/index.tsx [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+async function loadCourse(id: number) {
+  const params = {
+    courseId: id,
+  }
+
+  const course = await sanityClient.fetch(courseQuery, params)
+  return course
+}
+
+export async function getStaticProps() {
```
Comment (zacjones93) [inline src/pages/projects/build-modern-layouts-with-css-grid/index.tsx:243]: using the same `getStaticProps` pattern as before for sanity loading

**Review by zacjones93 — COMMENTED** [review]
Updated the PR with @laurosilvacom to have of a story and introduce the challenge in a human way with far more context than before

**Comment by laurosilvacom** [comment]
Feeling much better about the content! 

We update the content to showcase the purpose of the page and communicate the outcomes to the user. We provided linear instructions to the challenges and outlined expected outcomes.

**Review by zacjones93 — APPROVED** [review]

**Pull request merged**